helix-core/src/main/java/org/apache/helix/monitoring/mbeans/ClusterStatusMonitor.java

```diff
@@ -583,6 +643,31 @@ public void updateMissingTopStateDurationStats(String resourceName, long totalDu
     }
   }
 
+  public void updateMissingTopStateDurationThreshold(long missingTopStateDurationThreshold) {
+    _asyncMissingTopStateMonitor.setMissingTopStateDurationThreshold(missingTopStateDurationThreshold);
```

Comment:
now this is where i have question: since your _asyncMissingTopStateMonitor has synchronized and is in tight-loop, will this method go through? May be when the method is in "wait" state, it releases the lock and so this method can go through

Reply:
Answered in above comments. It's just not about in wait() state but whenever it's timesliced and this thread get it's timeslice it can call any non-synchronized methods of _asyncMissingTopStateMonitor.